prusti-contracts-common/Cargo.toml

```diff
@@ -0,0 +1,8 @@
+[package]
```

Comment:
Is this the start of a stdlib crate? Is it automatically used by Prusti somehow?

Reply:
Yes, a stdlib crate. Not used by Prusti itself, only in tests. The purse is so that a prusti user can import contracts for some common stdlib functions.

Please let me know if there is a better place for these contracts. Ideally, they would be in `prusti-contracts`, however, `prusti-contracts` does not use `stdlib` (and I guess there is a reason for it), and I somehow couldn't make prusti syntax work right in the `prusti-contracts` crate.

Reply:
Thank you for the suggestion; putting the contracts to `prusti-contracts-internal` would be great. However, when trying to `pub use extern_specs;` from the crate, I get the following error:
```
`proc-macro` crate types currently cannot export any items other than functions tagged with `#[proc_macro]`, `#[proc_macro_derive]`, or `#[proc_macro_attribute]`
```
is there something we can do to overcome this limitation, or does this mean we still need a new crate for the extern specs?

Reply:
Ah right, I forgot about that limitation for proc-macro crates :clown_face: Then that leaves two options I think:

1. A new crate for extern specs like you mentioned. This could then become a dependency of `prusti-contracts`, but needs some special treatment to avoid accidentally linking the stdlib when prusti is not enabled. I think something like this in `Cargo.toml` of `prusti-contracts` should work:

```toml
[dependencies]
prusti-contracts-extern-spec = { path = "../prusti-contracts-extern-spec", optional = true }

[features]
prusti = ["prusti-contracts-internal", "prusti-contracts-extern-spec"]
```

2. Alternatively, since `prusti-contracts` itself is not a proc-macro crate, you could create a Rust module `extern_specs` in `prusti-contracts` and then conditionally include it from `src/lib.rs` when the `prusti` feature is enabled:

```rust
// When the prusti feature is enabled, we need access to the
// stdlib in order to be able to write `#[extern_spec]` for it.
#![cfg_attr(not(feature = "prusti"), no_std)] // instead of the current `#![no_std]` at the top

// When prusti is active, include the module at `src/extern_spec.rs`
// or at `src/extern_spec/mod.rs`, which can in turn include other
// modules if you want to organize the extern specs into modules.
#[cfg(feature = "prusti")]
mod extern_specs; // `pub` keyword is not needed here, I think?
```

Reply:
Thanks, that's super useful @Pointerbender! I'm not sure option 1. would work, since the `prusti-contracts-extern-spec` would also need to have `prusti-contracts` as a dependency, but option 2. should work great - I'll try it out now